src/components/frame/v5/pages/AgreementsPage/FilterContext/FilterContext.tsx

```diff
@@ -0,0 +1,188 @@
+  const [dateFilters, setDateFilters] = useState<DateOptions>({
+    pastHour: false,
+    pastDay: false,
+    pastWeek: false,
+    pastMonth: false,
+    pastYear: false,
+    custom: undefined,
+  });
+
+  const handleMotionStatesFilterChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const isChecked = event.target.checked;
+      const name = event.target.name as MotionState;
+      const failedStatuses = [
+        'Failed',
+        'FailedNotFinalizable',
+      ] as MotionState[];
+
+      if (isChecked) {
+        if (name === 'Failed') {
+          setMotionStates([...motionStates, ...failedStatuses]);
+        } else {
+          setMotionStates([...motionStates, name]);
+        }
+      } else if (name === 'Failed') {
+        setMotionStates(
+          motionStates.filter(
+            (checkedItem) =>
+              checkedItem !== 'Failed' &&
+              checkedItem !== 'FailedNotFinalizable',
+          ),
+        );
+      } else {
+        setMotionStates(
+          motionStates.filter((checkedItem) => checkedItem !== name),
+        );
+      }
+    },
+    [motionStates],
+  );
+
+  const handleDateFilterChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const isChecked = event.target.checked;
+      const name = event.target.name as keyof DateOptions;
+
+      if (isChecked) {
+        setDateFilters({ ...dateFilters, [name]: true });
+      } else {
+        setDateFilters({ ...dateFilters, [name]: false });
+      }
+    },
+    [dateFilters],
+  );
+
+  const handleCustomDateFilterChange = useCallback(
+    (date: [Date | null, Date | null]) => {
+      const [newStartDate, newEndDate] = date;
+
+      if (newStartDate && newEndDate) {
+        setDateFilters({
+          ...dateFilters,
+          custom:
+            !newStartDate && !newEndDate
+              ? undefined
+              : [newStartDate.toString(), newEndDate.toString()],
+        });
+      }
+    },
+    [dateFilters],
+  );
+
+  const activeFilters: AgreementsPageFilters = useMemo(() => {
+    const date = getDateFilter(dateFilters);
+
+    return {
+      ...(motionStates.length ? { motionStates } : {}),
+      ...(searchFilter ? { search: searchFilter } : {}),
+      ...(date || {}),
+    };
+  }, [motionStates, searchFilter, dateFilters]);
+
+  const handleResetFilters = useCallback((category: string) => {
+    switch (category) {
+      case formatText({ id: 'agreementsPage.filter.status' }): {
+        return setMotionStates([]);
+      }
+      case formatText({ id: 'agreementsPage.filters.date' }):
+      case formatText({ id: 'agreementsPage.filters.date.custom' }): {
+        return setDateFilters({
+          pastHour: false,
+          pastDay: false,
+          pastWeek: false,
+          pastMonth: false,
+          pastYear: false,
+          custom: undefined,
+        });
+      }
+      default: {
+        return undefined;
+      }
+    }
+  }, []);
```

Comment:
The category should be an enum on its own, we shouldn't rely on translated strings as identifiers. 

You can format the enum value to a translated string right when you are displaying them.